pkg/dataobj/internal/sections/streams/iter.go

```diff
@@ -91,8 +95,13 @@ func IterSection(ctx context.Context, dec encoding.StreamsDecoder, section *file
 // Decode decodes a stream from a [dataset.Row], using the provided columns to
 // determine the column type. The list of columns must match the columns used
 // to create the row.
-func Decode(columns []*streamsmd.ColumnDesc, row dataset.Row) (Stream, error) {
-	var stream Stream
+func Decode(columns []*streamsmd.ColumnDesc, row dataset.Row, stream *Stream) error {
+	labelColumns := labelColumns(columns)
+	stream.Labels = slicegrow.GrowToCap(stream.Labels, labelColumns)
+	stream.Labels = stream.Labels[:labelColumns]
+	stream.LbValueCaps = slicegrow.GrowToCap(stream.LbValueCaps, labelColumns)
+	stream.LbValueCaps = stream.LbValueCaps[:labelColumns]
```

Comment:
It'd be nice to get rid of these for simplicity; is is possible for us to do something similar to what we did with `RecordMetadata` in `logs.Decode`?

Reply:
We can, but we use the same Stream object & labels.Labels functionality quite a bit in the `getOrAddStream` method (hash, equality, sorting, etc.) and I didn't want to replace them, at least not in this PR.
In the logs section we don't use that functionality so it was easy to replace labels.Labels with a custom RecordMetadata which references a []byte field.